simpeg/electromagnetics/time_domain/sources.py

```diff
@@ -197,37 +206,135 @@ class RampOffWaveform(BaseWaveform):
     >>> from simpeg.electromagnetics import time_domain as tdem
 
     >>> times = np.linspace(0, 1e-4, 1000)
-    >>> waveform = tdem.sources.RampOffWaveform(off_time=1e-5)
+    >>> waveform = tdem.sources.RampOffWaveform(1e-5)
     >>> plt.plot(times, [waveform.eval(t) for t in times])
     >>> plt.show()
 
     """
 
-    def __init__(self, off_time=0.0, **kwargs):
-        super().__init__(off_time=off_time, has_initial_fields=True, **kwargs)
+    def __init__(self, *args, ramp_start=None, ramp_end=None, **kwargs):
+        off_time = kwargs.pop("off_time", None)
+        if off_time is not None:
+            if len(args) > 1 or ramp_end is not None:
+                raise TypeError(
+                    "Can not specify both `off_time` and a `ramp_end` value."
+                )
+            else:
+                ramp_end = off_time
+                warnings.warn(
+                    "`off_time` keyword arg has been deprecated and will be removed in "
+                    "SimPEG v0.26.0, pass the ramp end time as the last positional argument.`",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
```

Comment:
Minor suggestion here. We don't need the `else` (I also included the change to v0.27.0 here):
```suggestion
            if len(args) > 1 or ramp_end is not None:
                raise TypeError(
                    "Can not specify both `off_time` and a `ramp_end` value."
                )
            ramp_end = off_time
            warnings.warn(
                "`off_time` keyword arg has been deprecated and will be removed in "
                "SimPEG v0.27.0, pass the ramp end time as the last positional argument.`",
                DeprecationWarning,
                stacklevel=2,
            )
```